ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -3094,3 +3095,53 @@ def test_ipahealthcheck_mismatching_certificates_subject(self):
         finally:
             # cleanup
             tasks.uninstall_master(self.master)
+
+
+class TestIPAHealthcheckWithCALess(CALessBase):
+    """
+    Install CALess server with user provided certificate.
+    """
+    num_replicas = 0
+
+    @classmethod
+    def install(cls, mh):
+        super(TestIPAHealthcheckWithCALess, cls).install(mh)
+        cls.create_pkcs12('ca1/server')
+        cls.prepare_cacert('ca1')
+        result = cls.install_server()
+        assert result.returncode == 0
+
+    @pytest.fixture
+    def expire_cert_warn(self):
+        """
+        Fixture to move the cert to about to expire, by moving the
+        system date using date -s command and revert it back
+        """
+        self.master.run_command(['date','-s', '+11Months10Days'])
+        yield
+        self.master.run_command(['date','-s', '-11Months10Days'])
+        self.master.run_command(['ipactl', 'restart'])
+
+    def test_ipahealthcheck_warns_on_expired_user_certs(self, expire_cert_warn):
+        """
+        This testcase checks that ipa-healthcheck warns
+        on expiring user-provided certificates.
+        """
+        msg = (
+            'Request id {key} expires in {days} days. '
+            'You need to manually renew this certificate'
```

Comment:
The message should end with a dot, otherwise the assert in line 3147 fails: `assert check["kw"]["msg"] == msg`